shell/platform/windows/direct_manipulation.cc

```diff
@@ -0,0 +1,309 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/logging.h"
+
+#include "flutter/shell/platform/windows/direct_manipulation.h"
+#include "flutter/shell/platform/windows/window_binding_handler_delegate.h"
+#include "flutter/shell/platform/windows/window_win32.h"
+
+namespace flutter {
+
+STDMETHODIMP DirectManipulationEventHandler::QueryInterface(REFIID iid,
+                                                            void** ppv) {
+  if ((iid == IID_IUnknown) ||
+      (iid == IID_IDirectManipulationViewportEventHandler)) {
+    *ppv = static_cast<IDirectManipulationViewportEventHandler*>(this);
+    AddRef();
+    return S_OK;
+  } else if (iid == IID_IDirectManipulationInteractionEventHandler) {
+    *ppv = static_cast<IDirectManipulationInteractionEventHandler*>(this);
+    AddRef();
+    return S_OK;
+  }
+  return E_NOINTERFACE;
+}
+
+HRESULT DirectManipulationEventHandler::OnViewportStatusChanged(
+    IDirectManipulationViewport* viewport,
+    DIRECTMANIPULATION_STATUS current,
+    DIRECTMANIPULATION_STATUS previous) {
+  if (current == DIRECTMANIPULATION_RUNNING) {
+    if (!resetting_) {
+      if (owner_->binding_handler_delegate) {
+        owner_->binding_handler_delegate->OnPointerFlowStart(
+            reinterpret_cast<int32_t>(this));
+      }
+    }
+  } else if (previous == DIRECTMANIPULATION_RUNNING) {
+    if (resetting_) {
+      resetting_ = false;
+    } else {
+      if (owner_->binding_handler_delegate) {
+        owner_->binding_handler_delegate->OnPointerFlowEnd(
+            reinterpret_cast<int32_t>(this));
+      }
+      // Need to reset the content transform
+      // Use resetting_ flag to prevent sending reset also to the framework
+      resetting_ = true;
```

Comment:
DirectManipulation is based around moving a "content" within a "viewport". We are sort of faking that here since our use case is not a simple content block, after each gesture the fake content needs to be reset to its original position. I'll add some explanation in the comments.